guac_core/src/storage/in_memory.rs

```diff
@@ -0,0 +1,147 @@
+use channel_client::types::{Channel, ChannelState};
+use channel_storage::ChannelStorage;
+use clarity::Address;
+use crypto::CryptoService;
+use failure::Error;
+use futures::future::ok;
+use futures::Future;
+use num256::Uint256;
+use qutex::Qutex;
+use std::collections::HashMap;
+
+/// A in-memory storage that stores data in
+pub struct InMemoryStorage {
+    // TODO: Optimize lookups
```

Comment:
Nope, use a hashmap and we can care about optimization later, like in years. Rust is by default fast enough for any data structure you make that isn't outright intentionally stupid.

Reply:
@jkilpatr I agree. It was a HashMap before (with ChannelId as the key when guac identified channels by ChannelIds). Identifying channels by Address or ChannelId when its available is unnecessary complication and it would simplify the flow a lot if we'd just use Address as key.